src/main/java/core/basesyntax/operation/PurchaseOperation.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.data.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        Storage.fruitStorage.put(
+                fruit,
+                Storage.fruitStorage.getOrDefault(fruit, 0) - quantity
```

Consider adding a check to ensure that the resulting quantity in storage does not become negative. If the purchase quantity exceeds the available stock, you might want to throw an exception or handle it appropriately.